### PR TITLE
Swift 3 beta 4 support

### PR DESCRIPTION
**Pull request by NachoSoto**
### TODOs:

- [x] Update `Quick` and `Nimble`.
- [x] Verify tests pass.
- [ ] Point `Result` back to `master`.
- [x] ~~Figure out [why the hell `Lock`, `Locking`, and `RecursiveLock` have `NS` prefixes again](https://twitter.com/NachoSoto/status/760722173891842052).~~: https://github.com/apple/swift-evolution/pull/421

### CODE REVIEW

**Comment by ikesyo** [comment]
> Figure out why the hell Lock, Locking, and RecursiveLock have NS prefixes again.

https://github.com/apple/swift-evolution/commit/bea5eab614b954775754639fb83a957a180152e1

**Comment by NachoSoto** [comment]
@ikeyso thanks: https://twitter.com/NachoSoto/status/760725120801538049
🙄

File: ReactiveCocoa/Swift/Scheduler.swift [hunk 1]
```diff
@@ -170,26 +170,14 @@ public final class QueueScheduler: DateSchedulerProtocol {
...
-		}
-
-		self.init(internalQueue: DispatchQueue(label: name, attributes: [.serial, qosAttribute]))
+	public convenience init(
+		qos: DispatchQoS = .default,
+		name: String = "org.reactivecocoa.ReactiveCocoa.QueueScheduler"
+	) {
+		self.init(internalQueue: DispatchQueue(
+			label: name,
+			qos: qos
```
Comment (NachoSoto) [inline ReactiveCocoa/Swift/Scheduler.swift]: This is much nicer now :)

**Comment by NachoSoto** [comment]
This build now, but I don't know if tests are passing until `Quick` and `Nimble` are updated. Looks like there are a few issues with them: https://github.com/norio-nomura/Nimble/pull/5.

**Comment by mdiep** [comment]
[Travis has ß4 now](https://twitter.com/travisci/status/760993848214421504). 😂

**Comment by andersio** [comment]
Is `DispatchQueue` by default a serial queue now? Seeing no `.serial` in `DispatchQueue.Attributes` is confusing...

**Comment by andersio** [comment]
Tests are passing now with a quick patch to Nimble. Quick doesn't need any patches.

**Comment by NachoSoto** [comment]
@andersio it's not a good practice to push commits to somebody else's PR. I think #2857 is an exception because it's an aggregation of many PRs, but this is not. I would have appreciated if you opened a separate PR targeting this, so that we could have reviewed that separately...

**Comment by NachoSoto** [comment]
Tests are passing! (Except for the flaky Objective-C test). Shall we merge?

**Pull request merged**